### ISSUE TITLE
Rollover API failing with 400 in case settings name don't have "index." prefix or "index" section in json

### ISSUE THREAD

**Issue by shwetathareja** [elastic/elasticsearch#53388]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.6

**Plugins installed**: []

**Description of the problem including expected versus actual behavior**:
Rollover API takes index settings as input parameters. The settings can specified as one of the below options: 
```
1.  "settings": {
      "index" : {
    	"number_of_shards": 2
      }
  }
 2.  "settings": {
    "index.number_of_shards": 2
  }
3. "settings": {
    "number_of_shards": 2
  }
```

But, with this [PR](https://github.com/elastic/elasticsearch/pull/50388). the #3 option doesn't work and throws error as

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "unknown setting [number_of_shards] did you mean [index.number_of_shards]?"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "unknown setting [number_of_shards] did you mean [index.number_of_shards]?"
  },
  "status" : 400
}
```

**Fix:**
There is normalize and settings validation logic in onlyCreateIndex method and after refactoring this code is not getting triggered during rollover flow which calls applyCreateIndexRequest method.
```
Settings.Builder updatedSettingsBuilder = Settings.builder();
        Settings build = updatedSettingsBuilder.put(request.settings()).normalizePrefix(IndexMetaData.INDEX_SETTING_PREFIX).build();
        indexScopedSettings.validate(build, true); // we do validate here - index setting must be consistent
        request.settings(build);
```

The normalize call should also run during rollover flow otherwise it breaks the expected behavior.

**Steps to reproduce**:

```
1. curl -XPUT localhost:9200/abc-000001
2. curl -XPUT localhost:9200/abc-000001/_alias/abc
3. curl -X POST "localhost:9200/abc/_rollover?pretty" -H 'Content-Type: application/json' -d'
{
"conditions": {
"max_age":   "1s"
},
"settings": {
"number_of_shards": 2
}
}
'
```
**Tried running this in 7.5 with no issue and rollover works as expected.**

**Comment by shwetathareja** [comment]
Hi Team,

If we are aligned on the issue and the fix. I can pick it up.

**Comment by andreidan** [comment]
Thanks for using ILM and reporting this bug @shwetathareja 
Feel free to work on a fix for this and let us know if you need any help.

**Comment by shwetathareja** [comment]
Sure andreidan@, Sorry for the delay. I will pick it up.

**Comment by Gaurav614** [comment]
Talked to @shwetathareja about this opened issue and wanted to pickup the implementation of it.
I will start working on this issue and will raise the PR for it shortly

**Comment by pugnascotia** [comment]
Closed by #57271.

**Comment by jbaiera** [comment]
Closing for real this time
